raiden/network/proxies/token_network_registry.py

```diff
@@ -113,107 +115,332 @@ def add_token(
+            if failed_receipt:
+                failed_at_blocknumber = failed_receipt["blockNumber"]
+
+                max_token_networks = self.get_max_token_networks(
+                    block_identifier=failed_at_blocknumber
+                )
+                token_networks_created = self.get_token_network_created(
+                    block_identifier=failed_at_blocknumber
+                )
+                already_registered = self.get_token_network(
+                    token_address=token_address, block_identifier=failed_at_blocknumber
+                )
+                deprecation_executor = self.get_deprecation_executor(
+                    block_identifier=failed_at_blocknumber
+                )
+                settlement_timeout_min = self.settlement_timeout_min(
+                    block_identifier=failed_at_blocknumber
+                )
+                settlement_timeout_max = self.settlement_timeout_max(
+                    block_identifier=failed_at_blocknumber
+                )
+                chain_id = self.get_chain_id(block_identifier=failed_at_blocknumber)
+                secret_registry_address = self.get_secret_registry_address(
+                    block_identifier=failed_at_blocknumber
+                )
+
+                if failed_receipt["cumulativeGasUsed"] == gas_limit:
+                    msg = (
+                        f"createERC20TokenNetwork failed and all gas was used "
+                        f"({gas_limit}). Estimate gas may have underestimated "
+                        f"createERC20TokenNetwork, or succeeded even though an assert is "
+                        f"triggered, or the smart contract code has an "
+                        f"conditional assert."
+                    )
+                    raise RaidenRecoverableError(msg)
+
+                if token_networks_created >= max_token_networks:
+                    raise RaidenRecoverableError(
+                        "The number of existing token networks reached the maximum allowed"
+                    )
+
+                if already_registered:
+                    # Race condition lost, the token network was created in a different
+                    # transaction which got mined first.
+                    raise RaidenRecoverableError(
+                        "The token was already registered in the TokenNetworkRegistry."
+                    )
+
+                if deprecation_executor == NULL_ADDRESS:
+                    raise RaidenUnrecoverableError(
+                        "The deprecation executor property for the "
+                        "TokenNetworkRegistry is invalid."
+                    )
+
+                if chain_id == 0:
+                    raise RaidenUnrecoverableError(
+                        "The chain ID property for the TokenNetworkRegistry is invalid."
+                    )
+
+                if chain_id != self.blockchain_service.network_id:
```

Comment:
Here too, this check is nowhere in smart contracts.  And I think Raiden should be usable on both chains after a fork.

Reply:
This point is kept track in https://github.com/raiden-network/raiden/pull/4956